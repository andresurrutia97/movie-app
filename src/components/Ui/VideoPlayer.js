import React from 'react';
import PropTypes from 'prop-types';
import ReactPlayer from 'react-player';

const VideoPlayer = ({ url }) => {
  return (
    <div className="h-full shadow-md">
      <ReactPlayer url={url} width="100%" height="100%" controls playing />
    </div>
  );
};

VideoPlayer.propTypes = {
  url: PropTypes.string.isRequired,
};

export default VideoPlayer;
