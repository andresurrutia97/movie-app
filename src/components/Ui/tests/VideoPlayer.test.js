import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ReactPlayer from 'react-player';

import VideoPlayer from '../VideoPlayer';

describe('<VideoPlayer />', () => {
  const component = <VideoPlayer url="videoUrl" />;

  it('should render correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain an instance of <ReactPlayer/> with the url equal to "videoUrl"', () => {
    const wrapper = mount(component);
    expect(wrapper.find(ReactPlayer)).toHaveLength(1);
    expect(wrapper.find(ReactPlayer).props().url).toBe('videoUrl');
  });
});
