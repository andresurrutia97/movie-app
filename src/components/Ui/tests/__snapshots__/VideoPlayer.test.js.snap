// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayer /> should render correctly 1`] = `
<div
  className="h-full shadow-md"
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;
